cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -796,15 +867,15 @@ func (h *bundleHandler) addApplication(change *bundlechanges.AddApplicationChang
 	// specialization.
 	if charm.CharmStore.Matches(chID.URL.Schema) {
 		var track string
-		if h.origin.Track != nil {
-			track = *h.origin.Track
+		if origin.Track != nil {
+			track = *origin.Track
 		}
 		platform, err := utils.DeducePlatform(cons, series, h.modelConstraints)
 		if err != nil {
 			return errors.Trace(err)
 		}
 		// A channel is needed whether the risk is valid or not.
-		channel, _ := corecharm.MakeChannel(track, h.origin.Risk, "")
+		channel, _ := corecharm.MakeChannel(track, origin.Risk, "")
```

Comment:
Will we get an invalid `channel` value if this returns an error? If so, we should handle the error.

Reply:
No, the `origin.Risk` has already been resolved much higher, see `getOrigin` and that way we should be safe to assume that we don't need to handle the error twice.